lib/buffer.js

```diff
@@ -64,7 +64,17 @@ function alignPool() {
  * much breakage at this time. It's not likely that the Buffer constructors
  * would ever actually be removed.
  **/
+var newBufferWarned = false;
 function Buffer(arg, encodingOrOffset, length) {
+  if (!new.target && !newBufferWarned) {
+    newBufferWarned = true;
+    process.emitWarning(
+      'Using Buffer without `new` will soon stop working. ' +
+      'Use `new Buffer()`, or preferably ' +
+      '`Buffer.from()`, `Buffer.allocUnsafe()` or `Buffer.alloc()` instead.',
```

Comment:
Is your concern that users will needlessly use `Buffer.allocUnsafe`? If so, I don't think the order matters much. The "Unsafe" part is already a good indicator that it's not recommended.

What seems a bit more likely (although still very unlikely) is that if we change the order to `from`, `alloc`, `allocUnsafe`, the user will stop reading after `alloc`, prompting a situation described in https://github.com/nodejs/node/pull/8169#discussion_r75487236.